server/svix-server/src/v1/endpoints/endpoint.rs

```diff
@@ -208,11 +216,25 @@ impl From<EndpointHeaders> for EndpointHeadersOut {
     }
 }
 
-impl ModelIn for EndpointHeadersIn {
+#[derive(Clone, Debug, PartialEq, Validate, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct EndpointHeadersPatchIn {
+    #[validate]
+    headers: EndpointHeaders,
+}
+
+impl ModelIn for EndpointHeadersPatchIn {
     type ActiveModel = endpoint::ActiveModel;
 
     fn update_model(self, model: &mut Self::ActiveModel) {
-        model.headers = Set(Some(self.headers));
+        let existing = mem::replace(&mut model.headers, Set(None)).unwrap();
```

Comment:
Not gonna lie, I don't remember what `mem::replace` does, but it feels very wrong to use `mem`. What are you trying to do here?

Reply:
I agree it seems strange, but `mem::replace` is the lowest-overhead way I know of to modify a non-`Copy` field from a mutably borrowed struct: https://doc.rust-lang.org/error-index.html#E0507. 

The simplest alternative I know is to clone the field but was trying to avoid it.

Reply:
Yeah, definitely shouldn't copy the structure, but why wouldn't something like this work?
```rust
        model.headers = match model.headers {
            Some(&mut existing) => {
                existing.0.extend(self.headers.0);
                Set(Some(existing))
            }
            None => Set(Some(self.headers)),
        };
```
As in without that line? Why do you even need to touch model.headers before?

Reply:
If I use something like above:
```rust
        model.headers = match model.headers {
            Some(existing) => {
                existing.0.extend(self.headers.0);
                Set(Some(existing))
            }
            None => Set(Some(self.headers)),
        };
```

I get the move error. Let me go back to the drawing board for a bit and see if there is a cleaner way.

Reply:
What are you unwrapping? Also, I think this is actually something that also changed in axum (though maybe in the unreleased version).

Reply:
`model.headers` is a `sea_orm::ActiveValue` -- unwrapping if I want direct access to the underlying EndpointHeaders struct. I'm trying to see if I can do the pattern matching on the ActiveValue itself rather than the unwrapped value.

Reply:
Yeah, just do pattern matching there too, or maybe you can use `take()`.
```rust
        model.headers = match model.headers.take() {
            Some(existing) => {
                existing.0.extend(self.headers.0);
                Set(Some(existing))
            }
            None => Set(Some(self.headers)),
        };
```

Reply:
Done.